pytensor/graph/basic.py

```diff
@@ -1646,6 +1774,27 @@ def is_in_ancestors(l_apply: Apply, f_apply: Apply) -> bool:
     return False
 
 
+def variable_is_in_ancestors(
+    node: Variable, check: Union[Variable, Collection[Variable]]
+) -> bool:
+    """Determine if `node` is in the graph given by any in check.
+    Parameters
+    ----------
+    node: Variable
+        Node to check
+    check: Collection[Variable]
+        Nodes to check dependency on
+    Returns
+    -------
+    bool
+    """
+    if not isinstance(check, Collection):
+        check = {check}
+    else:
+        check = set(check)
+    return any(interim in check for interim in ancestors([node], blockers=check))
```

Comment:
I would inline this function, I think it's simple enough for users to figure out. We can add it later if there's a demand for it.

Reply:
I would like to merge this with `is_in_ancestors` so it is more polymorphic

```python
NodeT can be wither Variable or Apply 
is_in_ancestors(node[NodeT], search: List[NodeT]) -> bool
```

Reply:
What I was saying is I don't think we need this helper function at all. Ancestors gives you an iterable and you check if it is in it. I don't see why we need a user-facing function for the one-liner

Reply:
Okay, we already had one for nodes, so that opens a valid precedent for variables. I think we should keep separate. There's a subtle difference between checking for nodes and variables.

Reply:
I renamed the is_in_ancestors to apply_in_ancestors and unified the api with variable_in_ancestors so they are semantically the same. Should we bother with deprecation warnings and add a fallback?